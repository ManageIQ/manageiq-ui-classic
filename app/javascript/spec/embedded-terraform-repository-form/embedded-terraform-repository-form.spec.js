import React from 'react';
import toJson from 'enzyme-to-json';
import fetchMock from 'fetch-mock';
import { shallow } from 'enzyme';
import EmbeddedTerraformRepositoryForm from '../../components/embedded-terraform-repository-form';

require('../helpers/set_fixtures_helper.js');
require('../helpers/old_js_file_require_helper.js');
require('../helpers/miqSparkle.js');
require('../helpers/miqAjaxButton.js');

jest.mock('../../helpers/miq-redirect-back', () => jest.fn());

describe('Embedded Terraform Repository form component', () => {
  let submitSpyMiqSparkleOn;
  let submitSpyMiqSparkleOff;

  const attributes = ['name', 'description', 'scm_url', 'verify_ssl', 'authentication_id', 'scm_branch'];

  const repositoryMock = {
    authentication_id: null,
    description: 'workflows for testing',
    href: 'http://localhost:3000/api/configuration_script_sources/46',
    id: '46',
    name: 'embedded_terraform_repository_test',
    scm_branch: '',
    scm_url: 'github.com:ManageIQ/workflows-examples',
    verify_ssl: 0,
  };

  beforeEach(() => {
    submitSpyMiqSparkleOn = jest.spyOn(window, 'miqSparkleOn');
    submitSpyMiqSparkleOff = jest.spyOn(window, 'miqSparkleOff');
  });

  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
    submitSpyMiqSparkleOn.mockRestore();
    submitSpyMiqSparkleOff.mockRestore();
  });

  it('should render add form', (done) => {
    const wrapper = shallow(<EmbeddedTerraformRepositoryForm />);

    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });

  it('should render edit variant', async(done) => {
    fetchMock.get(`/api/configuration_script_sources/46?attributes=${attributes.join(',')}`, repositoryMock);
    const wrapper = shallow(<EmbeddedTerraformRepositoryForm repositoryId="46" />);

    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });

  it('should submit add form', async(done) => {
    const wrapper = shallow(<EmbeddedTerraformRepositoryForm />);
    const data = {
      authentication_id: '',
      description: 'workflows for testing',
      name: 'embedded terraform repository add test',
      scm_branch: '',
      scm_url: 'github.com:ManageIQ/workflows-examples',
      verify_ssl: true,
      manager_provider: { href: 'http://localhost:3000/api/providers/77' },
    };
    fetchMock.postOnce('/api/configuration_script_sources/', data);
    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });

  it('should submit edit form', async(done) => {
    const wrapper = shallow(<EmbeddedTerraformRepositoryForm />);
    fetchMock.get(`/api/configuration_script_sources/46?attributes=${attributes.join(',')}`, repositoryMock);
    const data = {
      authentication_id: null,
      description: 'workflows for testing',
      name: 'embedded terraform repository edit test',
      scm_branch: '',
      scm_url: 'github.com:ManageIQ/workflows-examples',
      verify_ssl: false,
    };
    fetchMock.postOnce('/api/configuration_script_sources/46', data);
    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });
});
