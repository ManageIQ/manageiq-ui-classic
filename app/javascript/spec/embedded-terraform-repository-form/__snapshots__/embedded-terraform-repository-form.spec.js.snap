// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Embedded Terraform Repository form component should render add form 1`] = `
<Loading
  active={true}
  className="export-spinner"
  description="Active loading indicator"
  small={true}
  withOverlay={false}
/>
`;

exports[`Embedded Terraform Repository form component should render edit variant 1`] = `
<Loading
  active={true}
  className="export-spinner"
  description="Active loading indicator"
  small={true}
  withOverlay={false}
/>
`;

exports[`Embedded Terraform Repository form component should submit add form 1`] = `
<Loading
  active={true}
  className="export-spinner"
  description="Active loading indicator"
  small={true}
  withOverlay={false}
/>
`;

exports[`Embedded Terraform Repository form component should submit edit form 1`] = `
<Loading
  active={true}
  className="export-spinner"
  description="Active loading indicator"
  small={true}
  withOverlay={false}
/>
`;
